### ISSUE TITLE
Annotation cards can be interacted with the mouse but not with the keyboard

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#6306]
One challenge to achieve this is the fact that annotation cards are not focusable at the moment, so we don't have a way to announce this shortcut, but we could probably change this, or at least make them work as long as any inner element is focused.

A similar pattern can be observed at sites like Twitter. In there, every tweet is part of the focus sequence, and they have `aria-labelledby` pointing to the tweet's text, author and a few other things, `role="article"` and `tabindex="0"`.

Some time ago, we tried to [make annotation cards focusable](https://github.com/hypothesis/client/pull/5539), but we had to end up [reverting it](https://github.com/hypothesis/client/pull/5572) as it caused [other problems](https://github.com/hypothesis/support/issues/31) (more context can be found on our internal slack https://hypothes-is.slack.com/archives/C4K6M7P5E/p1687440646316049).

We should have this in mind if we attempt to solve this again.

> Related issue: https://github.com/hypothesis/product-backlog/issues/1425
> Some comments include alternative approaches.
